lib/expand/join.go

```diff
@@ -98,6 +101,8 @@ type Peer struct {
 	// server is the gRPC installer server
 	server *server.Server
 
+	// abortC is a channel to signal external operation abort
+	abortC     chan chan struct{}
```

Comment:
I needed this channel to be able to guarantee delivery of the abort message to the client (but only if there's a client).